docs/collecting-container-logs.md

```diff
@@ -0,0 +1,170 @@
+# Collecting container logs
+<!-- TOC -->
+- [Configuration](#configuration)
+  - [Multiline log parsing](#multiline-log-parsing)
+  - [Setting source name and other built-in metadata](#setting-source-name-and-other-built-in-metadata)
+  - [Filtering](#filtering)
+  - [Modifying log records](#modifying-log-records)
+  - [Persistence](#persistence)
+- [Advanced Configuration](#advanced-configuration)
+  - [Advanced Filtering](#advanced-filtering)
+  - [Disabling container logs](#disabling-container-logs)
+<!-- /TOC -->
+
+By default, log collection is enabled. This includes both container logs and systemd logs. This document covers container logs.
+
+Container logs are read and parsed directly from the Node filesystem, where the kubelet writes them under the `/var/log/pods`
+directory. They are then sent to a metadata enrichment service which takes care of adding Kubernetes metadata, custom processing,
+filtering, and finally sending the data to Sumo Logic. Both the collection and the metadata enrichment are done by the OpenTelemetry Collector.
+
+See the [Solution Overview diagram](README.md#log-collection) for a visualisation.
+
+## Configuration
+
+High level  configuration for logs is located in [`values.yaml`][values] under the `sumologic.logs` key. Configuration
+specific to container logs is located under the `sumologic.logs.container` key.
+
+Configuration specific to the log collector DaemonSet can be found under the `otellogs` key.
+
+Finally, configuration specific to the metadata enrichment StatefulSet can be found under the `metadata.logs` key.
+
+### Multiline log parsing
+
+By default, each line output by an application is treated as a separate log record. However, some applications can actually
+output logs split into multiple lines - this is often the case for stack traces, for example. If we want such a multiline log
+to appear in Sumo Logic as a single record, we need to tell the collector how to distinguish between lines which start a new record
+and ones which continue an existing record.
+
+Multiline log parsing can be configured using the `sumologic.logs.multiline` section in `user-values.yaml`.
+
+```yaml
+sumologic:
+  logs:
+    multiline:
+      enabled: true
+      first_line_regex: "^\\[?\\d{4}-\\d{1,2}-\\d{1,2}.\\d{2}:\\d{2}:\\d{2}"
+```
+
+where `first_line_regex` is a regular expression used to detect the first line of a multiline log.
+
+This feature is enabled by default. It can rarely cause problems by merging together lines which are supposed to be separate.
+In that case, feel free to disable it.
```

Comment:
It would be nice to give example logs which are being catched by this regexp

Reply:
Done